an_devices/advanced_navigation_device_serial.py

```diff
@@ -28,56 +28,58 @@
 ################################################################################
 
 import os
-import datetime
 import serial
 import serial.serialutil as serialutil
-from enum import Enum
-from struct import pack, unpack
-from dataclasses import dataclass, field
-from ctypes.wintypes import DOUBLE
-from anpp_packets.packets.an_packet_protocol import AN_Decoder
 from abc import ABC, abstractmethod
-from anpp_packets.packets.anpp_packets import ResetVerification, OdometerFlags, ExternalAirDataPacketFlags, PacketPeriod, GPIOOutputRate
 
-valid_baud_rates = [2400, 4800, 9600, 19200, 38400, 57600,
-                    115200, 230400, 250000, 460800, 500000,
-                    800000, 921600, 1000000, 1250000, 2000000]
+from anpp_packets.an_packets import PacketID
+from anpp_packets.an_packet_protocol import ANDecoder
+from anpp_packets.an_packet_1 import RequestPacket
 
-class AdvancedNavigationDevice(ABC):
-    def __init__(self, port, baud, log = False):
-        self.bytes_waiting = AN_Decoder()
+
+class AdvancedNavigationDeviceSerial(ABC):
+    def __init__(self, port, baud):
+        self.decoder = ANDecoder()
         self.ser = None
-        self.logFile = None
 
         if isinstance(port, str):
             self.port = port
         else:
-            print(f"Port:{port} is not valid")
+            raise ValueError(f"Port:{port} is not valid")
 
-        if (int(baud) in valid_baud_rates):
+        if int(baud) in self.valid_baud_rates:
             self.baud = baud
```

Comment:
Should baud be stored as an int or expected to be passed as an int? This code could take a string that is convertible to a valid int and store it as a string. ie
```suggestion
        if isinstance(baud, int) and baud in self.valid_baud_rates:
            self.baud = baud
```
or
```suggestion
        if int(baud) in self.valid_baud_rates:
            self.baud = int(baud)
```